drivers/ad7746/ad7746.c

```diff
@@ -0,0 +1,448 @@
+
+int ad7746_read_voltage_vdd(ad7746_t *dev, int *value)
+{
+    return _read_voltage_temp(dev, value, AD7746_VT_MD_VDD);
+}
+
+int ad7746_read_temperature_int(ad7746_t *dev, int *value)
+{
+    return _read_voltage_temp(dev, value, AD7746_VT_MD_TEMP);
+}
+
+int ad7746_read_temperature_ext(ad7746_t *dev, int *value)
+{
+    return _read_voltage_temp(dev, value, AD7746_VT_MD_ETEMP);
+}
+
+int ad7746_set_cap_ch_input(const ad7746_t *dev, ad7746_cap_input_t input)
+{
+    uint8_t reg;
+    int res = AD7746_NOI2C;
+
+    assert(dev);
+    i2c_acquire(I2C);
+
+    if (i2c_read_reg(I2C, ADDR, AD7746_REG_CAP_SETUP, &reg, 0)) {
+        goto release;
+    }
+
+    if (input == AD7746_CAP_IN_2) {
+        reg |= (1 << AD7746_CAP_SETUP_CIN2_BIT);
+    }
+    else {
+        reg &= ~(1 << AD7746_CAP_SETUP_CIN2_BIT);
+    }
+
+    if (i2c_write_reg(I2C, ADDR, AD7746_REG_CAP_SETUP, reg, 0)) {
+        goto release;
+    }
+
+    res = AD7746_OK;
+
+release:
+    i2c_release(I2C);
+    return res;
+}
+
+int ad7746_set_cap_sr(const ad7746_t *dev, ad7746_cap_sample_rate_t sr)
+{
+    uint8_t reg;
+    int res = AD7746_NOI2C;
+
+    assert(dev);
+    i2c_acquire(I2C);
+
+    if (i2c_read_reg(I2C, ADDR, AD7746_REG_CONFIGURATION, &reg, 0)) {
+        goto release;
+    }
+
+    reg &= ~(7 << AD7746_CONFIGURATION_CAPF0_BIT);
+    reg |= (sr << AD7746_CONFIGURATION_CAPF0_BIT);
+
+    if (i2c_write_reg(I2C, ADDR, AD7746_REG_CONFIGURATION, reg, 0)) {
+        goto release;
+    }
+
+    res = AD7746_OK;
+
+release:
+    i2c_release(I2C);
+    return res;
+}
+
+int ad7746_set_vt_sr(const ad7746_t *dev, ad7746_vt_sample_rate_t sr)
+{
+    uint8_t reg;
+    int res = AD7746_NOI2C;
+
+    assert(dev);
+    i2c_acquire(I2C);
+
+    if (i2c_read_reg(I2C, ADDR, AD7746_REG_CONFIGURATION, &reg, 0)) {
+        goto release;
+    }
+
+    reg &= ~(3 << AD7746_CONFIGURATION_VTF0_BIT);
+    reg |= (sr << AD7746_CONFIGURATION_VTF0_BIT);
+
+    if (i2c_write_reg(I2C, ADDR, AD7746_REG_CONFIGURATION, reg, 0)) {
+        goto release;
+    }
+
+    res = AD7746_OK;
+
+release:
+    i2c_release(I2C);
+    return res;
+}
+
+static int _read_capacitance(ad7746_t *dev, int *value,
+                             ad7746_cap_input_t input)
+{
+    int res;
+    uint32_t raw;
+
+    if (dev->params.cap_input != input) {
+        if (ad7746_set_cap_ch_input(dev, input)) {
+            return AD7746_NOI2C;
+        }
+        /* discard last reading */
+        _read_raw_ch(dev, AD7746_READ_CAP_CH, &raw);
+    }
+
+    res = _read_raw_ch(dev, AD7746_READ_CAP_CH, &raw);
+    if (res == AD7746_OK) {
+        *value = _raw_to_capacitance(raw);
+    }
+
+    return res;
+}
+
+static int _read_voltage_temp(ad7746_t *dev, int *value, ad7746_vt_mode_t mode)
+{
+    int res;
+    uint32_t raw;
+
+    if (dev->params.vt_mode != mode) {
+        if (ad7746_set_vt_ch_mode(dev, mode)) {
+            return AD7746_NOI2C;
+        }
+        xtimer_usleep(_vt_sr_times[dev->params.vt_sample_rate] * US_PER_MS);
```

Comment:
Spend a comment that we sleep for one conversion cycle and flush the first sample in order to get stable output. Maybe mention that we cribbed from the linux implementation.